src/compiler/checker.ts

```diff
@@ -765,7 +765,23 @@ namespace ts {
             }
         }
 
-        function getJsxNamespace(): __String {
+        function getJsxNamespace(location: Node | undefined): __String {
+            if (location) {
+                const file = getSourceFileOfNode(location);
+                if (file) {
+                    if (file.localJsxNamespace) {
+                        return file.localJsxNamespace;
+                    }
+                    const jsxPragma = file.pragmas.get("jsx");
+                    if (jsxPragma) {
+                        const chosenpragma = isArray(jsxPragma) ? jsxPragma[0] : jsxPragma;

```

Comment:
why not set this one in `processPragmasIntoFields` like the other ones, just for consistency.

Reply:
It pretty much maps to `localJsxFactory`, it's calculation is just deferred.

Reply:
I could move the calculation from the checker into the parser, if you'd like.